apps/metadata-editor/src/app/edit/components/top-toolbar/top-toolbar.component.ts

```diff
@@ -12,9 +16,37 @@ import { MatIconModule } from '@angular/material/icon'
     PublishButtonComponent,
     ButtonComponent,
     MatIconModule,
+    TranslateModule,
   ],
   templateUrl: './top-toolbar.component.html',
   styleUrls: ['./top-toolbar.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TopToolbarComponent {}
+export class TopToolbarComponent {
+  protected SaveStatus = [
+    'saved_not_published',
+    'saved_published_up_to_date',
+    'saved_published_changes_pending',
+    // these are not used since the draft is saved locally
+    // TODO: use these states when the draft is saved on the server
+    // 'saving',
+    // 'saving_failed',
+  ] as const
+
+  protected saveStatus$: Observable<typeof this.SaveStatus[number]> =
+    combineLatest([
+      this.editorFacade.alreadySavedOnce$,
+      this.editorFacade.changedSinceSave$,
+    ]).pipe(
+      map(([alreadySavedOnce, changedSinceSave]) => {
+        if (!alreadySavedOnce) {
+          return 'saved_not_published'
+        }
+        return changedSinceSave
+          ? 'saved_published_changes_pending'
+          : 'saved_published_up_to_date'
+      })
+    )
```

Comment:
I'm a bit confused about the save status and already saved once here. Are we talking about the same "save" (draft on localStorage vs record on server)?
As I understand it, the draft is always saved, hence all the status beginning with "saved_".
But the record might not be saved, hence the need to call "alreadySavedOnce".
Could we maybe switch to a word like "stored" instead of "saved" for the localStorage?

Reply:
This is also a good point, it can be confusing. I guess we could change the statuses like so:
* "saved as draft only" => when creating a record
* "record is up-to-date" => when the record was just published (ie saved on the server)
* "draft saved, changes pending" => when the record was modified and not yet published

Reply:
I find those new labels much better! Will you change the technical statuses names too?

Reply:
yep :)